src/commit.c

```diff
@@ -476,6 +490,17 @@ int git_commit__parse(void *_commit, git_odb_object *odb_obj)
 	return -1;
 }
 
+int git_commit__parse(void *_commit, git_odb_object *odb_obj)
+{
+	git_commit *commit = _commit;
+
+	int error = git_commit__parse_ext(commit, odb_obj, GIT_COMMIT_PARSE_ALL);
+	if (error < 0)
+		return error;
+
+	return 0;
+}
```

Comment:
Is there a case where `git_commit__parse_ext` will return a positive value that needs to be treated as success?  Reading this function in isolation, I wondered if perhaps `parse_ext` returned the length of the buffer parsed or something.  It appears to not, and this could be simplified as a one-liner `return git_commit__parse_ext...`

Not saying you need to do that, but I'm just trying to clarify my understanding, I guess, by typing into this little box.